PowerShell/JumpCloud Module/Public/Reports/Get-JCReport.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+.Synopsis
+Ordered list of report metadata
+.Description
+Ordered list of report metadata
+.Example
+PS C:\> Get-JCReport
+
+Returns a list of all available reports
+.Example
+PS C:\> Get-JCReport -Sort 'CREATED_AT'
+
+Returns a list of all available reports, sorted by the most recently created report
+
+.Outputs
+JumpCloud.SDK.DirectoryInsights.Models.IGet200ApplicationJsonItemsItem
+.Link
+https://github.com/TheJumpCloud/jcapi-powershell/tree/master/SDKs/PowerShell/JumpCloud.SDK.DirectoryInsights/docs/exports/Get-JcSdkReport.md
+#>
+Function Get-JCReport {
+    [OutputType([JumpCloud.SDK.DirectoryInsights.Models.IGet200ApplicationJsonItemsItem])]
+    [CmdletBinding(DefaultParameterSetName = 'List', PositionalBinding = $false)]
+    Param(
+        [Parameter()]
+        [ArgumentCompleter([JumpCloud.SDK.DirectoryInsights.Support.Sort])]
+        [JumpCloud.SDK.DirectoryInsights.Category('Query')]
+        [JumpCloud.SDK.DirectoryInsights.Support.Sort]
+        # Sort type and direction.
+        # Default sort is descending, prefix with - to sort ascending.
+        ${Sort},
+        [Parameter(ValueFromPipelineByPropertyName, ParameterSetName = 'Report', HelpMessage = 'ID of the Report request.')]
+        [String]$ReportID,
```

Comment:
We need to manually change the parameter sets. In our acceptance criteria, we had talked about an "all" set and a "byID" set. 

I was initially confused when I just tried to run the function like `Get-JCReport -ReportID "someID"` the command will fail and return a 404, which is correct, because the `artifactID` is missing. 

If we decide to keep code this way, the `ArtifactID` and `ReportID` parameters need to be set to `mandatory=$true`. 

But I'd argue we should potentially remove the `artifactID` and replace it with a `type` parameter for `json` or `csv`. When I'm using this function, I'd want to just return the data instead of writing a loop to check if the report is finished before copying the artifactID — the `Get-JCReport` function can do that for us.